rmw_opensplice_cpp/src/rmw_node.cpp

```diff
@@ -294,4 +387,180 @@ rmw_destroy_node(rmw_node_t * node)
   rmw_node_free(node);
   return result;
 }
+
+rmw_ros_meta_t * 
+rmw_get_node_names(void)
+{  
+  // Use the current ROS_DOMAIN_ID.
+  char * ros_domain_id = nullptr;
+  const char * env_var = "ROS_DOMAIN_ID";
+#ifndef _WIN32
+  ros_domain_id = getenv(env_var);
+#else
+  size_t ros_domain_id_size;
+  _dupenv_s(&ros_domain_id, &ros_domain_id_size, env_var);
+#endif  
+  size_t domain_id = std::stoi(ros_domain_id);
+
+  // On Windows, setting the ROS_DOMAIN_ID does not fix the problem, so error early.
+#ifdef _WIN32
+  if (!ros_domain_id) {
+    RMW_SET_ERROR_MSG("environment variable ROS_DOMAIN_ID is not set");
+    fprintf(stderr, "[rmw_opensplice_cpp]: error: %s\n", rmw_get_error_string_safe());
+    return nullptr;
+  }
+#endif
+
+  // Create a ROS middleware node that will be used to 
+  // fetch other participant's data from the "ros_meta" topic
+  rmw_node_t * rmw_node;
+  rmw_node = rmw_create_node("get_node_names", domain_id);
```

Comment:
I think we should avoid creating an internal node just to get the metadata. I think it would be better to have each node do this even if that's less efficient in the short term. We can work on avoiding duplicated effort within a single process later by using a daemon or some intelligent shared resource.